pageserver/src/tenant/timeline.rs

```diff
@@ -4149,6 +4149,12 @@ impl Timeline {
         cancel: &CancellationToken,
         ctx: &RequestContext,
     ) -> anyhow::Result<()> {
+        let _timer = self
+            .metrics
+            .update_gc_info_histo
+            .start_timer()
+            .record_on_drop();
```

Comment:
I think `record_on_drop` is a mistake here. If there's a broken tenant in the future that bails out early, it will skew the histogram.

ALso, I expect we'll have a multimodal histogram in the regular case anyways.
One peak near zero with pitr_interval=0 because that caode path is super cheap.
Then one peak at pitr_interval=1d (free tier).
And another peak at pitr_interval=7d (default for paid plans).

Why different peaks? Because `find_lsn_for_timestamp` runtime depends on gc_horizon, which is determined primarily by pitr_interval.

Reply:
Agreed, it will have multiple peaks. However, I thought the per-timeline values (or the per-tenant sum) would still be useful to catch durations in which branch creation will be slow.